plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/ComparePresenter.java

```diff
@@ -75,45 +78,106 @@ public ComparePresenter(AppContext appContext,
     }
 
     /**
-     * Show compare window.
+     * Show compare window for given set of files between given revision and latest code version.
      *
-     * @param file
-     *         file name with its full path
-     * @param status
-     *         status of the file
+     * @param changedItems
+     *         ordered touched files
+     * @param currentFile
+     *         file which will be shown first, if null then the first from the list will be shown
      * @param revision
      *         hash of revision or branch
      */
-    public void showCompareWithLatest(final File file, final Status status, final String revision) {
-        this.comparedFile = file;
+    public void showCompareWithLatest(final ChangedItems changedItems,
+                                      @Nullable final String currentFile,
+                                      final String revision) {
+        this.changedItems = changedItems;
         this.revision = revision;
+
         this.compareWithLatest = true;
+        view.setEnableSaveChangesButton(true);
+
+        setupCurrentFile(currentFile);
+        showCompareForCurrentFile();
+    }
+
+    /**
+     * Shows compare window for given set of files between specified revisions.
+     *
+     * @param changedItems
+     *         ordered touched files
+     * @param currentFile
+     *         file which will be shown first, if null then the first from the list will be shown
+     * @param revisionA
+     *         hash of the first revision or branch.
+     *         If it is set to {@code null}, compare with empty repository state will be performed
+     * @param revisionB
+     *         hash of the second revision or branch.
+     *         If it is set to {@code null}, compare with latest repository state will be performed
+     */
+    public void showCompareBetweenRevisions(final ChangedItems changedItems,
+                                            @Nullable final String currentFile,
+                                            @Nullable final String revisionA,
+                                            @Nullable final String revisionB) {
+        this.changedItems = changedItems;
+        this.revisionA = revisionA;
+        this.revisionB = revisionB;
+
+        this.compareWithLatest = false;
+        view.setEnableSaveChangesButton(false);
+
+        setupCurrentFile(currentFile);
+        showCompareForCurrentFile();
+    }
+
+    /**
+     * Shows comparison for selected file.
+     * Type of comparison to show depends on {@code compareWithLatest} field.
+     */
+    private void showCompareForCurrentFile() {
+        view.setEnableNextDiffButton(currentItemIndex != (changedItems.getFilesQuantity() - 1));
+        view.setEnablePreviousDiffButton(currentItemIndex != 0);
+
+        changedItems.getProject()
+                    .getFile(changedItems.getItemByIndex(currentItemIndex))
+                    .then(file -> {
+                        if (file.isPresent()) {
```

Comment:
if `!file.isPresent()` it means it has been deleted. So, we have to show diff also.